server/auth/handlers/handler.go

```diff
@@ -337,11 +338,13 @@ func (h *AuthHandlers) enrichTmplData(req *request.AuthReq) interface{} {
 	if req.AuthUser != nil {
 		// fetch current user with updated fields
 		user, err := h.UserService.FindByAny(req.Context(), req.AuthUser.User.ID)
-		if err != nil {
-			return nil
+
+		// check if err is not nil and cater for MFA by checking if the error is not allowed to read
+		if err != nil && !errors.Is(err, service.UserErrNotAllowedToRead()) {
+			return err
 		}
 
-		d["user"] = user
+		d["user"] = req.AuthUser.User
 		d["theme"] = user.Meta.Theme
```

Comment:
is there a reason you did not change to `req.AuthUser` here as well?

Reply:
Yes, the reason is `req.AuthUser` doesn't have the most updated data.